test/tests/functional/pbs_nonprint_characters.py

```diff
@@ -741,7 +741,7 @@ def test_nonprint_character_interactive_job(self):
             # Once all commands sent and matched, job exits
             self.server.expect(JOB, 'queue', op=UNSET, id=jid, offset=1)
             # Check for the non-printable character in the output file
-            with open(self.job_out1_tempfile) as fd:
+            with open(self.job_out1_tempfile, newline="") as fd:
```

Comment:
Do we need the same change at lines 786 and 859?

Reply:
No we don't because those parts test terminal control characters and \r is not used.